### PR TITLE
use code dependencies

### PR DESCRIPTION
**Pull request by hannesa2**
Otherwise, samples are using old libraries all the time.
I made in `HelloIOIO` a flavor to switch dependencies to test. This could be a baseground to test in future jitpack.io dpendencies

### CODE REVIEW

File: software/applications/HelloIOIO/build.gradle [hunk 1]
```diff
@@ -6,11 +6,26 @@ android {
...
+    }
+
 }
 
 dependencies {
-    implementation 'com.github.ytai.ioio:IOIOLibAndroid:5.07'
-    implementation 'com.github.ytai.ioio:IOIOLibAndroidBluetooth:5.07'
-    implementation 'com.github.ytai.ioio:IOIOLibAndroidAccessory:5.07'
-    implementation 'com.github.ytai.ioio:IOIOLibAndroidDevice:5.07'
+    codeImplementation project(':IOIOLibAndroidBluetooth')
```
Comment (topherbuckley) [inline software/applications/HelloIOIO/build.gradle:23]: What is this call? I'm not familiar with this and don't see it in the gradle [docs](https://docs.gradle.org/current/userguide/java_library_plugin.html)

File: software/applications/HelloIOIO/build.gradle [hunk 2]
```diff
@@ -6,11 +6,26 @@ android {
...
+    codeImplementation project(':IOIOLibAndroidAccessory')
+    codeImplementation project(':IOIOLibAndroidDevice')
+
+    aarImplementation 'com.github.ytai.ioio:IOIOLibAndroid:5.07' // the old artifacts still need it
```
Comment (topherbuckley) [inline software/applications/HelloIOIO/build.gradle:27]: This call as well.

Comment (hannesa2) [reply software/applications/HelloIOIO/build.gradle:27]: It's an `Implementation` of flavor `aar`

Comment (hannesa2) [reply software/applications/HelloIOIO/build.gradle:27]: If you have flavor `aar`  you use `aarImplementation`
With flavor `code`  you use `codeImplementation`

**Comment by topherbuckley** [comment]
Just going through and trying to identify trimable branches as you mentioned a while back you did not want a lot of dead branches hanging around. Any reason this branch has not been deleted?

**Comment by hannesa2** [comment]
> Just going through and trying to identify trimable branches as you mentioned a while back you did not want a lot of dead branches hanging around. Any reason this branch has not been deleted?

image

Maybe you should run `git remote prune origin` or `git remote prune upstream`

**Comment by topherbuckley** [comment]
Yep! Sorry for the noise. I did that an deleted my comment shortly thereafter only to see you are a speedy reader and had already replied.

**Pull request merged**